src/layouts/Sidebar.jsx

```diff
@@ -39,6 +41,55 @@ const Sidebar = ({ menu, activeMenuItem }) => {
     setActiveState({ ...activeState, sub: subLinkId });
   };
 
+  useEffect(() => {
+    axios.get(`${API_END_POINT}/api/list/batch/`, { headers }).then((res) => {
+      setBatchList(res.data.data);
+    });
+  }, []);
+
+  const [showInputFields, setShowInputFields] = useState(false);
+
+  const [isAddingBatch, setIsAddingBatch] = useState(false); // New state for switching text
+
+  const toggleInputFields = () => {
+    setShowInputFields((prevState) => !prevState); // Toggle input fields visibility directly using prevState
+    setIsAddingBatch(!showInputFields);
+
+    // setBatchList(true)// Update text directly based on the current state of input fields
+  };
+
```

Comment:
We might change adding batch inside side bar later need to discuss with @Idrish . Remind me tomorrow.
If we're providing ability to add batch , then we have to remove this code (state and handlers)